### PR TITLE
[FLINK-30859] Remove all Kafka connector code from main repo

### PR DESCRIPTION
**Pull request by mas-chen**
~This is WIP, DO NOT MERGE!~

### CODE REVIEW

File: flink-architecture-tests/pom.xml [hunk 1]
```diff
@@ -177,7 +177,7 @@ under the License.
 			<dependency>
 				<groupId>org.apache.flink</groupId>
 				<artifactId>flink-connector-kafka</artifactId>
-				<version>${project.version}</version>
+				<version>3.0.0-1.17</version>
```
Comment (zentol) [inline flink-architecture-tests/pom.xml:180]: This dependency should be removed, with the connector defining it's own prod/test architecture tests (similar to what we did for other connectors).

**Review by zentol — COMMENTED** [review]
There are some Kafka checkstyle suppressions in `tools/maven/suppressions.xml`.

**Comment by mas-chen** [comment]
@flinkbot run azure

**Comment by mas-chen** [comment]
@flinkbot run azure

**Review by tzulitai — APPROVED** [review]
Thanks for following up on this @mas-chen. Did another pass on the removals - on the whole, we haven't removed stuff that shouldn't be removed or doesn't exist in the flink-connector-kafka repo, and the build is passing. So +1, LGTM.

There's a few places (mostly tests and examples) which we can improve to not rely on Kafka, but let's do that as a follow up.

I'll proceed to merge this now.

**Comment by tzulitai** [comment]
Merged via 149a5e34c1ed8d8943c901a98c65c70693915811

**Comment by davidradl** [comment]
@zentol @tzulitai As the kafka connector code has now been removed from the repo in the change that has been merged in [#22228](https://github.com/apache/flink/commit/149a5e34c1ed8d8943c901a98c65c70693915811)  , this and the other referenced Kafka connector orientated prs need to be moved to the new Flink Kafka connector repository.

**Comment by davidradl** [comment]
@zentol @tzulitai As the kafka connector code has now been removed from the repo in the change that has been merged in [#22228](https://github.com/apache/flink/commit/149a5e34c1ed8d8943c901a98c65c70693915811)  , this and the other referenced Kafka connector orientated prs need to be moved to the new Flink Kafka connector repository.

**Pull request closed without merging**